hack/generated/pkg/genruntime/conditions/conditions.go

```diff
@@ -0,0 +1,166 @@
+	ConditionSeverityNone ConditionSeverity = ""
+)
+
+type ConditionType string
+
+const (
+	// ConditionTypeReady is a condition indicating if the resource is ready or not.
+	// A ready resource is one that has been successfully provisioned to Azure according to the
+	// resource Spec.
+	ConditionTypeReady = "Ready"
+)
+
+var _ fmt.Stringer = Condition{}
+
+// Condition defines an extension to status (an observation) of a resource
+type Condition struct {
+	// Type of condition.
+	// +kubebuilder:validation:Required
+	Type ConditionType `json:"type"`
+
+	// Status of the condition, one of True, False, or Unknown.
+	// +kubebuilder:validation:Required
+	Status metav1.ConditionStatus `json:"status"`
+
+	// Severity with which to treat failures of this type of condition.
+	// For conditions which have positive polarity (Status == True is their normal/healthy state), this will be omitted when Status == True
+	// For conditions which have negative polarity (Status == False is their normal/healthy state), this will be omitted when Status == False.
+	// This is omitted in all cases when Status == Unknown
+	// +kubebuilder:validation:Optional
+	Severity ConditionSeverity `json:"severity,omitempty"`
+
+	// LastTransitionTime is the last time the condition transitioned from one status to another.
+	// +kubebuilder:validation:Required
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+
+	// Reason for the condition's last transition.
+	// Reasons are upper CamelCase (PascalCase) with no spaces. A reason is always provided, this field will not be empty.
+	// +kubebuilder:validation:Required
+	Reason string `json:"reason"`
+
+	// Message is a human readable message indicating details about the transition. This field may be empty.
+	// +kubebuilder:validation:Optional
+	Message string `json:"message,omitempty"`
+}
+
+// IsEquivalent returns true if this condition is equivalent to the passed in condition.
+// Two conditions are equivalent if all of their fields EXCEPT LastTransitionTime are the same.
+func (c Condition) IsEquivalent(other Condition) bool {
+	return c.Type == other.Type &&
+		c.Status == other.Status &&
+		c.Severity == other.Severity &&
+		c.Reason == other.Reason &&
+		c.Message == other.Message
+}
+
+// Copy returns an independent copy of the Condition
+func (c Condition) Copy() Condition {
+	return c
+}
+
+// String returns a string representation of this condition
+func (c Condition) String() string {
+	return fmt.Sprintf(
+		"Condition [%s], Status = %q, Severity = %q, Reason = %q, Message = %q, LastTransitionTime = %q",
+		c.Type,
+		c.Status,
+		c.Severity,
+		c.Reason,
+		c.Message,
+		c.LastTransitionTime)
+}
+
+// SetCondition sets the provided Condition on the Conditioner. The condition is only
+// set if the new condition is in a different state than the existing condition of
+// the same type.
+func SetCondition(o Conditioner, new Condition) {
+	if o == nil {
+		return
+	}
+
+	conditions := o.GetConditions()
```

Comment:
Since it's our interface, why not have `GetConditions()` return a `map[ConditionType]Condition` that allows for easy updates/checks?

Reply:
If we return a `map` is that we can't fuss with ordering if we want - which I think we might end up wanting/needing to do in order to make updates work well (CAPI does it). Obviously we could do ordering inside of the interface implementation but that means we end up duplicating the ordering code in a lot of implementations.

Given that (for better or worse) the actual property on status is a slice, I think it makes sense for the interface to expose that lowest level primitive. If we want to write a wrapper that uses that interface to get the slice and then convert it to a map, work with that and then convert it back, we can already do that above the level of the interface (which saves us having to write code generation to do the to/from map logic in each impl).

One more (quite minor) consideration is that I was modeling this after the CAPI implementation (which we also referenced for the `Condition` shape) and their interface looks (mostly) like this

Reply:
maybe let's sync on this in person?

I'm not understanding how what you're suggesting solves for example the ordering problem I mentioned, nor why we necessarily need it in this interface rather than in a wrapper if we want. The actual data is a slice and has to be a slice (because that's what the Kubernetes conditions spec requires) so any time we act on a map we must convert it back to a slice at the end.

Reply:
Synced up in person and I'll go ahead and make the change as discussed

Reply:
Looks good.